test/conformance/conformance_flowgraph.h

```diff
@@ -59,32 +62,523 @@ struct first_functor {
+}
+
+template<typename Node, typename ...Args>
+void test_priority(Args... args){
+    std::size_t concurrency_limit = 1;
+    oneapi::tbb::global_control control(oneapi::tbb::global_control::max_allowed_parallelism, concurrency_limit);
+
+    oneapi::tbb::flow::graph g;
+
+    oneapi::tbb::flow::continue_node<oneapi::tbb::flow::continue_msg> source(g,
+                                          [](oneapi::tbb::flow::continue_msg){ return oneapi::tbb::flow::continue_msg();});
+
+    conformance::first_functor<int>::first_id = -1;
+    conformance::first_functor<int> low_functor(1);
+    conformance::first_functor<int> high_functor(2);
+
+    Node high(g, args..., high_functor, oneapi::tbb::flow::node_priority_t(1));
+    Node low(g, args..., low_functor);
+
+    make_edge(source, low);
+    make_edge(source, high);
+
+    source.try_put(oneapi::tbb::flow::continue_msg());
+
+    g.wait_for_all();
+    
+    CHECK_MESSAGE((conformance::first_functor<int>::first_id == 2), "High priority node should execute first");
+}
+
+template<typename Node, typename ...Args>
+void test_concurrency(Args...){
+    oneapi::tbb::global_control c(oneapi::tbb::global_control::max_allowed_parallelism,
+                                          oneapi::tbb::this_task_arena::max_concurrency());
+
+    int max_num_threads = tbb::global_control::active_value(
+        tbb::global_control::max_allowed_parallelism
+    );
+
+    std::vector<int> threads_count = {1, oneapi::tbb::flow::serial, max_num_threads, oneapi::tbb::flow::unlimited};
+
+    if(max_num_threads > 2){
+        threads_count.push_back(max_num_threads / 2);
+    }
+
+    for(auto num_threads : threads_count){
+        utils::ConcurrencyTracker::Reset();
+        int expected_threads = num_threads;
+        if(num_threads == oneapi::tbb::flow::unlimited){
+            expected_threads = max_num_threads;
+        }
+        oneapi::tbb::flow::graph g;
+        conformance::barrier_body counter(expected_threads);
+        Node fnode(g, num_threads, counter);
+
+        conformance::test_push_receiver<int> sink(g);
+
+        make_edge(fnode, sink);
+
+        for(int i = 0; i < 500; ++i){
+            fnode.try_put(i);
+        }
+        g.wait_for_all();
+    }
+}
+
+template<typename Node, typename ...Args>
+void test_rejecting(Args...){
+    oneapi::tbb::flow::graph g;
+
+    worker_body<int> body;
+    Node fnode(g, oneapi::tbb::flow::serial, body);
+
+    conformance::test_push_receiver<int> sink(g);
+
+    make_edge(fnode, sink);
+
+    bool try_put_state;
+
+    for(int i = 0; i < 10; ++i){
+        try_put_state = fnode.try_put(i);
+    }
+
+    g.wait_for_all();
+    CHECK_MESSAGE((conformance::get_values(sink).size() == 1), "Messages should be rejected while the first is being processed");
+    CHECK_MESSAGE((!try_put_state), "`try_put()' should returns `false' after rejecting");
+}
```

Comment:
It is a pseudocode, but I think the the common idea should be clear:

```suggestion
template<typename Node, typename ...Args>
void test_rejecting(Args...){
    oneapi::tbb::flow::graph g;

    barrier_body<continue_msg> body;
    Node fnode(g, oneapi::tbb::flow::serial, body);

    conformance::test_push_receiver<int> sink(g);

    make_edge(fnode, sink);

    bool try_put_state = fnode.try_put(cont_msg{});
    CHECK(try_put_state);
    try_put_state = fnode.try_put(cont_msg{});
    CHECK(!try_put_state);
    barrier_body<continue_msg>::flag = true;

    g.wait_for_all();
    CHECK_MESSAGE((conformance::get_values(sink).size() == 1), "Messages should be rejected while the first is being processed");
}
```